service/src/main/java/uk/nhs/adaptors/gpc/consumer/utils/MdcUtil.java

```diff
@@ -0,0 +1,19 @@
+package uk.nhs.adaptors.gpc.consumer.utils;
+
+import org.slf4j.MDC;
+import org.springframework.http.HttpHeaders;
+import org.springframework.web.server.ServerWebExchange;
+
+public class MdcUtil {
+    private static final String REQUEST_ID = "RequestId";
+    private static final String TRACE_ID = "Ssp-TraceID";
+
+    public static void applyHeadersToMdc(ServerWebExchange exchange) {
+        MDC.put(REQUEST_ID, exchange.getLogPrefix());
+        HttpHeaders headers = exchange.getRequest().getHeaders();
+        var traceId = headers.get(TRACE_ID);
+        if (traceId != null && traceId.size() > 0) {
```

Comment:
Lists have a method for this
```suggestion
        if (traceId != null && !traceId.isEmpty()) {
```